Assets/Plugins/iOS/Shopify/ApplePayEventDispatcher.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  ApplePayEventDispatcher.swift
+//  UnityBuySDK
+//
+//  Created by Shopify.
+//  Copyright © 2017 Shopify Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+import PassKit
+
+typealias AppleEventCompletion = (_ response: ApplePayEventResponse?) -> Void
+
+class ApplePayEventDispatcher: NSObject {
+    
+    let receiverName: String
+    
+    enum MethodName: String {
+        case updateSummaryItemsForShippingIdentifier = "UpdateSummaryItemsForShippingIdentifier"
+        case updateSummaryItemsForShippingContact    = "UpdateSummaryItemsForShippingContact"
+        case fetchCheckoutStatusForToken             = "FetchApplePayCheckoutStatusForToken"
+        case didFinishCheckoutSession                = "DidFinishCheckoutSession"
+    }
+    
+    // ----------------------------------
+    //  MARK: - Init -
+    //
+    init(recieverName: String) {
+        self.receiverName = recieverName
+        super.init()
+    }
+    
+    func call(method: MethodName, withValue value: String, completionHandler: AppleEventCompletion?) {
```

Comment:
This method signature can be optimized for call site clarity:
```swift
func call(_ method: Method, value: String, completionHandler: AppleEventCompletion? = nil)
```
Example call site:
```swift
call(.didFinishCheckoutSession, value: status.description)
```